### PR TITLE
Ft order status 31

### PR DESCRIPTION
**Pull request by Kevin-ishimwe**
#### What does this PR do?
adds order status tracking capabilities to backend
#### Description of Task to be completed?
- endpoint for getting order status 
- endpoint for updating order status
- websocket event listener 

#### How should this be manually tested?
open the repository documentation 
on `localhost:4000/docs` and test under the 
Order Details section

### CODE REVIEW

File: src/controllers/order/orderStatusController.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+});
+export const updateOrderStatus = asyncWrapper(async (req, res) => {
+  const id = req.params.id;
+  const { status } = req.body;
+  const order = await db.Order.update(
+    { status: status },
+    { where: { id: id } }
+  );
+  // Notify the client over a WebSocket connection that the order status has been updated
+  res.json({ status: req.t('success'), message: req.t('status_order') });
```
Comment (FabCode67) [inline src/controllers/order/orderStatusController.js]: add status code on response

File: src/controllers/order/orderStatusController.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { asyncWrapper } from '../../utils/handlingTryCatchBlocks';
+import db from '../../database/models';
+
+export const getOrderStatus = asyncWrapper(async (req, res) => {
+  const order = await db.Order.findOne({ where: { userId: req.user.id } });
+  // Send the order status to the client over a WebSocket connection
+  res.json({ status: req.t('success'), order: order });
```
Comment (FabCode67) [inline src/controllers/order/orderStatusController.js]: what status code here?

File: src/controllers/order/orderStatusController.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { asyncWrapper } from '../../utils/handlingTryCatchBlocks';
+import db from '../../database/models';
+
+export const getOrderStatus = asyncWrapper(async (req, res) => {
+  const order = await db.Order.findOne({ where: { userId: req.user.id } });
```
Comment (FabCode67) [inline src/controllers/order/orderStatusController.js:5]: what can happen if order is not found?

**Review by FabCode67 — CHANGES_REQUESTED** [review]

File: src/locales/fr/translation.json [hunk 4]
```diff
@@ -25,69 +25,68 @@
...
-  "productCategoryNotfound":"Catégorie de produit non trouvée.",
+  "productRetieved": "Produits récupérés avec succès",
+  "productCategoryNotfound": "Catégorie de produit non trouvée.",
   "Missing_query": "requete de recherche manquante",
   "searching": "complete avec succès!",
 
   "product_category_not_found": "Catégorie de produit introuvable.",
   "is_required": "est requis",
-  "product_does_not_exist_in_collection" : "Product doesn't exist in your collection",
+  "product_does_not_exist_in_collection": "Product doesn't exist in your collection",
```
Comment (FabCode67) [inline src/locales/fr/translation.json:54]: translate this line as expected

**Review by FabCode67 — CHANGES_REQUESTED** [review]

**Review by Evaristekanova — CHANGES_REQUESTED** [review]
you didn't specify status codes for the returned responses and i see your french transilation are still in English

File: src/controllers/order/orderStatusController.js (hunk 3 above)
Comment (Kevin-ishimwe) [reply src/controllers/order/orderStatusController.js:5]: it returns an empty array

File: src/locales/fr/translation.json (hunk 4 above)
Comment (Kevin-ishimwe) [reply src/locales/fr/translation.json:54]: this isnt in my work

**Review by FabCode67 — APPROVED** [review]

**Review by BenoitIT — APPROVED** [review]

**Review by Evaristekanova — APPROVED** [review]

**Pull request merged**